src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -540,12 +535,34 @@ class IpexCommunicationService extends AgentService {
       `${ConfigurationService.env.keri.credentials.testServer.urlInt}/oobi/${schemaSaid}`
     );
     const schema = await this.props.signifyClient.schemas().get(schemaSaid);
-    await this.ipexMessageStorage.createIpexMessageRecord({
+
+    let prefix;
+    let key;
+    switch (historyType) {
+    case ConnectionHistoryType.CREDENTIAL_REVOKED:
+      prefix = KeriaContactKeyPrefix.HISTORY_REVOKE;
+      key = message.exn.e.acdc.d;
+      break;
+    case ConnectionHistoryType.CREDENTIAL_ISSUANCE:
+    case ConnectionHistoryType.CREDENTIAL_REQUEST_AGREE:
+    case ConnectionHistoryType.CREDENTIAL_REQUEST_PRESENT:
+      prefix = KeriaContactKeyPrefix.HISTORY_IPEX;
+      key = message.exn.d;
+      break;
+    default:
+      throw new Error("Invalid history type");
+    }
+    const ipexHistory: IpexHistoryItem = {
       id: message.exn.d,
       credentialType: schema?.title,
       content: message,
       connectionId: message.exn.i,
       historyType,
+      timestamp: new Date().toISOString(),
```

Comment:
Hmm, sorry just noticed a few more things here:
- The timestamp is in `message.exn.dt` which is more accurate (in case we were offline etc)
- We don't need to store the whole message as `content` as it's no longer a basic record. The `id` is enough.
- It's not always `Ipex` related - e.g. the revocation could be noticed without IPEX. `ConnectionHistoryItem` would make more sense

So maybe:
```
const historyItem: ConnectionHistoryItem = {
  id: message.exn.d,
  dt: message.exn.dt,
  connectionId: message.exn.i,
  historyType,
  credentialType: schema.title
};
```

Reply:
ok, let me update